src/main/java/mate/academy/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Inject
+    private UserDao userDao;
+
+    @Override
+    public User add(User user) {
+        user.setSalt(HashUtil.getSalt());
+        user.setPassword(HashUtil.hashPassword(user.getPassword(), user.getSalt()));
+        return userDao.add(user);
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {

```

Comment:
Services should not return Optional, they handle null case

Reply:
![image](https://github.com/mate-academy/hibernate-user-service-hw/assets/49767049/cb4ea088-7c57-4866-89fc-0612ec130151)